File: programmers/코딩테스트 고득점 Kit/해시/의상.py
def solution(clothes) :
    closet = {}
    for cloth in clothes :
        num = hash(cloth[1])
        closet[num] = 0
    for cloth in clothes :
        num = hash(cloth[1])
        closet[num] += 1
    answer = 1
    for _, i in closet.items() :
        answer *= (i + 1)
    return answer - 1

File: programmers/코딩테스트 고득점 Kit/해시/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> map = new HashMap<>();
        for (String[] clothe: clothes){
            String type = clothe[1];
            map.put(type, map.getOrDefault(type, 0) + 1);
        }
        for (int temp : map.values()){
            answer *= (temp + 1);
        }
        return answer - 1;
    }
}
